hw3/03-form.js

```diff
@@ -1,10 +1,76 @@
-const http = require('http');
+const http = require("http");
+var qs = require("querystring");
 const port = process.env.PORT || 5001;
 
 // http://localhost:5001/form should return a form with input elements for username, email, and submit button
 
 // http://localhost:5001/submit should return all the data the user entered
 
+//This should be in a separate HTML file
+const postHTML = `<html><body>
+<form method='post'>
+<label for="username">Userame: </label>
+<input type="text" name="username" id="username"><br />
+<label for="email">Email: </label>
+<input type="text" name="email" id="email"><br />
+<label for="comments">Comments: </label>
+<input type="text" name="comments" id="comments"><br />
+<input type="radio" name="newsletter">
+<label for="newsletter">Sign up for the newsletter</label><br>
+<input type='submit' value="Submit">
+</form></body></html>`;
+let post = "";
+
+const server = http.createServer((req, res) => {
+  if (req.url === "/") {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    res.write(`<h1>Exercise 03</h1>`);
+    res.write(`<li><a href="/form">/form</a></li>`);
+    res.write(`<li><a href="/submit">/submit</a></li>`);
+    res.end();
+  } else if (req.url === "/form") {
+    let body = "";
+    req.on("data", (chunk) => {
+      body += chunk;
+      console.log("BODY on data: " + body);
+      post = qs.parse(body);
+      console.log("POST on /form " + post.username);
+    });
+    req.on("end", () => {
+      res.writeHead(200, { "Content-Type": "text/html" });
+      res.write(postHTML);
+      res.end();
+    });
+  } else if (req.url === "/submit") {
+    if (post.username !== undefined) {
+      res.write(`<p>Username: ${post.username}</p>`);
```

Comment:
Your code looks good, but I think there are some good changes you could make in this last exercise. It is much simpler and efficient to make an html file and serve it with the `server.static` library, using the `serveFile()` function. I also think since your parse() function returns a collection, you can OR the `post.username` with 'n/a' like this: `const name = post.username || 'n/a'` and it will save the name if it exists, or n/a if it doesn't.